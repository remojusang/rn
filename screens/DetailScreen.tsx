import React, { useEffect } from 'react';
import ScreenLayout from '../components/ScreenLayout';
import type { NativeStackScreenProps } from '@react-navigation/native-stack';
import { Text } from 'react-native';
import { RootStackParamList } from '../utils/types';
import { Pressable } from 'react-native';
import TabIcon from '../components/TabIcon';
import { useSetRecoilState } from 'recoil';
import { personDataState } from '../components/Atoms';

type DetailProps = NativeStackScreenProps<
  RootStackParamList,
  'Detail'
>;

function DetailScreen({ route, navigation }: DetailProps) {
  const setPersonData = useSetRecoilState(personDataState);
  useEffect(() => {
    if (route.params?.name) {
      navigation.setOptions({
        title: route.params.name,
      });
    }
  }, [navigation, route]);
  const handlePress = () => {
    setPersonData(prev =>
      prev.filter(({ name }) => route.params.name !== name),
    );
    navigation.navigate('Home');
  };
  return (
    <ScreenLayout isLoading={false}>
      <Text>name: {route?.params?.name}</Text>
      <Text>age: {route?.params?.age}</Text>
      <Text>gender: {route?.params?.gender}</Text>
      <Pressable
        style={{
          justifyContent: 'center',
          alignItems: 'center',
          width: 60,
          height: 60,
        }}
        onPress={handlePress}>
        <TabIcon name="trash" color="red" size={40} isFocused />
      </Pressable>
    </ScreenLayout>
  );
}

export default DetailScreen;
