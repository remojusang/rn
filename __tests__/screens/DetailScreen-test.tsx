import { render } from '@testing-library/react-native';
import DetailScreen from '../../screens/DetailScreen';
const ErrorBoundary = require('../../components/ErrorBoundary');

// Couldn't find a navigation object. Is your component inside NavigationContainer? 에러 방지 코드
// https://spin.atomicobject.com/2021/02/24/react-navigation-5-unit-testing-components/
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
    }),
  };
});

describe('DetailScreen', () => {
  test('렌더링 테스트', () => {
    const TEST_PROPS: any = {
      route: jest.fn(),
      navigation: jest.fn(),
    };
    const renderedJson = render(
      <ErrorBoundary>
        <DetailScreen {...TEST_PROPS} />,
      </ErrorBoundary>,
    ).toJSON();
    expect(renderedJson).toMatchSnapshot();
    expect(renderedJson).toBeTruthy();
  });
});
