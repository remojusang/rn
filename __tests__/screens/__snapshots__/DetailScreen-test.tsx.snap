// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailScreen 렌더링 테스트 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(50,50,50,1)",
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <Text>
    name: 
  </Text>
  <Text>
    age: 
  </Text>
  <Text>
    gender: 
  </Text>
</View>
`;
